src/rgw/rgw_admin.cc

```diff
@@ -4255,14 +4260,26 @@ int main(int argc, const char **argv)
       return EIO;
     }
 
+    cfgstore = DriverManager::create_config_store(dpp(), config_store_type);
+    if (!cfgstore) {
+      std::cerr << "Unable to initialize config store." << std::endl;
+      exit(1);
+    }
+    auto r = site.load(dpp(), null_yield, cfgstore.get());
+    if (r < 0) {
+      std::cerr << "Unable to initialize config store." << std::endl;
+      exit(1);
+    }
```

Comment:
we need to avoid calling this for "raw storage ops" like the realm/zonegroup/zone/period commands, because those don't require that we have a valid multisite configuration - they're necessary to set up a valid configuration in the first place

i dug up an old branch where i'd started on this. it needs a rebase, but it pushed it to https://github.com/ceph/ceph/compare/main...cbodley:ceph:wip-rgw-admin-site-config?expand=1 for reference. commit https://github.com/ceph/ceph/commit/933c978c116c716ca7568156aaaa31c666c43a2d shows where we can safely initialize this

because the "raw storage ops" won't have or expect a valid SiteConfig, i prefer that as a properly-scoped local variable instead of a global

Reply:
I think I have fixed the `radosgw-admin` issues. I can bring up multisite. I didn't follow the example from your commit, since we need the SiteConfig before we can initialize the driver, and it looks like we need the driver early on for option processing.